### PR TITLE
Add minor enhancements to `track/event` payload

### PR DESCRIPTION
**Pull request by jtapia**
## What does this change?
- Remove unnecessary code
- Add email validation on `track/event` calls

### CODE REVIEW

File: lib/solidus_klaviyo/configuration.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module SolidusKlaviyo
   class Configuration
-    attr_accessor :api_key, :public_key, :default_list, :email
```
Comment (jtapia) [inline lib/solidus_klaviyo/configuration.rb:5]: just a clean up, we don't need the `publick_key` anymore

File: spec/solidus_klaviyo_spec.rb [hunk 2]
```diff
@@ -40,8 +40,6 @@
     it 'updates the profile on the given list' do
       allow(described_class.configuration)
           .to receive(:api_key).and_return('test_key')
-      allow(described_class.configuration)
```
Comment (jtapia) [inline spec/solidus_klaviyo_spec.rb:43]: same for the tests

File: lib/solidus_klaviyo/tracker.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ def from_config
 
     # This method is called by SolidusTracking on `track` call
     def track(event)
+      return unless event.customer_properties.try(:[], '$email').presence
```
Comment (jtapia) [inline lib/solidus_klaviyo/tracker.rb:18]: add `email` validation for `SolidusTracking` interaction

File: lib/solidus_klaviyo/tracker.rb [hunk 4]
```diff
@@ -47,6 +49,8 @@ def track(event)
 
     # This method is called directly to send custom information
     def create_event(event_name, email, properties = {})
+      return unless email.presence
```
Comment (jtapia) [inline lib/solidus_klaviyo/tracker.rb:52]: add `email` validation for `SolidusKlaviyo` interaction

File: lib/solidus_klaviyo/configuration.rb (hunk 1 above)
Comment (jakepenn) [reply lib/solidus_klaviyo/configuration.rb:5]: nice!

**Pull request merged**